core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -2323,15 +2326,56 @@ class ReplicaManager(val config: KafkaConfig,
     }
     logManager.handleLogDirFailure(dir)
 
-    if (sendZkNotification)
+    if (notifyController) {
+      if (config.migrationEnabled) {
+        fatal(s"Shutdown broker because some log directory has failed during migration mode: $dir")
+        Exit.halt(1)
```

Comment:
This seems wrong as our long-term solution, but I guess it's OK for now. (We can discuss more later I guess)